### PR TITLE
[REVIEW] Add cudf::unique_count(table_view)

### PR DESCRIPTION
**Pull request by karthikeyann**
closes https://github.com/rapidsai/cudf/issues/5314
- [x] add unique_count(table_view)
- [x] add unit tests
  - [x] TableWithoutNull
  - [x] TableWithNull
  - [x] EmptyColumnedTable
  - [x] TableMixedTypes
  - [x] TableWithStringColumnWithNull

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/include/cudf/stream_compaction.hpp [hunk 1]
```diff
@@ -259,5 +259,19 @@ cudf::size_type unique_count(column_view const& input,
                              null_policy null_handling,
                              nan_policy nan_handling);
 
+/**
+ * @brief Count the unique elements in the table_view
+ *
+ * Given an input table_view, number of unique rows in this table_view is returned
+ *
+ * @param[in] input The column_view whose unique elements will be counted.
```
Comment (jrhemstad) [inline cpp/include/cudf/stream_compaction.hpp:267]:
```suggestion
 * @param[in] input Table whose unique rows will be counted.
```

File: cpp/include/cudf/stream_compaction.hpp [hunk 2]
```diff
@@ -259,5 +259,19 @@ cudf::size_type unique_count(column_view const& input,
...
+ * @param[in] nulls_equal flag to denote nulls are equal if null_equality::EQUAL,
```
Comment (jrhemstad) [inline cpp/include/cudf/stream_compaction.hpp:268]: No need to repeat the meaning of `null_equality::EQUAL/UNEQUAL`. 
```suggestion
 * @param[in] nulls_equal flag to denote if null elements should be considered equal
```

File: cpp/include/cudf/stream_compaction.hpp [hunk 3]
```diff
@@ -259,5 +259,19 @@ cudf::size_type unique_count(column_view const& input,
                              null_policy null_handling,
                              nan_policy nan_handling);
 
+/**
+ * @brief Count the unique elements in the table_view
+ *
+ * Given an input table_view, number of unique rows in this table_view is returned
```
Comment (jrhemstad) [inline cpp/include/cudf/stream_compaction.hpp:265]: This doesn't add anything that the `brief` doesn't already say. 
```suggestion
```

**Pull request merged**